src/features/supervisor-app/hooks/supervisor-app.ts

```diff
@@ -116,14 +115,30 @@ async function checkSupervisorReleaseUpgrades(
 		},
 	});
 
-	for (const release of releases) {
+	let doingUpgrade = false;
+
+	for (const release of oldReleases) {
 		const oldVersion = release.release_version;
-		if (semver.lt(newSupervisorVersion, oldVersion)) {
+		if (semver.gt(newSupervisorVersion, oldVersion)) {
+			doingUpgrade = true;
+		} else if (semver.lt(newSupervisorVersion, oldVersion)) {
 			throw new BadRequestError(
 				`Attempt to downgrade supervisor, which is not allowed`,
 			);
 		}
 	}
+
+	return doingUpgrade;
+}
+
+async function getRelease(api: sbvrUtils.PinejsClient, releaseId: number) {
```

Comment:
Can you please type the return value? This will then let you drop AnyObject from a couple of places.